include/boost/config/stdlib/libcpp.hpp

```diff
@@ -67,4 +67,14 @@
 // libc++ uses a non-standard messages_base
 #define BOOST_NO_STD_MESSAGES
 
+#if defined(__has_include)
+#if !__has_include(<shared_mutex>)
+#  define BOOST_NO_CXX14_HDR_SHARED_MUTEX
+#endif
+#elif defined(__clang__) && (__clang_major__ < 3 || (__clang_major__ == 3 && __clang_minor__ < 4))
```

Comment:
Don't use version check for clang. It's vulnerable as different vendors of clang use their own version numbering system. 

I think you would need this
```
#if defined(__has_include)
#  if !__has_include(<shared_mutex>)
#    define BOOST_NO_CXX14_HDR_SHARED_MUTEX
#  elif __cplusplus <= 201103L
#    define BOOST_NO_CXX14_HDR_SHARED_MUTEX
#  endif
#else
#  define BOOST_NO_CXX14_HDR_SHARED_MUTEX
#endif
```